components/toggle_modal_button_redux/toggle_modal_button_redux.jsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class ModalToggleButtonRedux extends React.Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        modalId: PropTypes.string.isRequired,
+        dialogType: PropTypes.func.isRequired,
+        dialogProps: PropTypes.object,
+        onClick: PropTypes.func,
+        className: PropTypes.string,
+        actions: PropTypes.shape({
+            openModal: PropTypes.func.isRequired,
+            closeModal: PropTypes.func.isRequired
+        }).isRequired
+    };
+
+    static defaultProps = {
+        dialogProps: {},
+        className: ''
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.show = this.show.bind(this);
+        this.hide = this.hide.bind(this);
```

Comment:
For coding standards, we prefer not to have binding like these and would just implement below functions like ``show = (e) => {...}``